rotkehlchen/data/import_utils.py

```diff
@@ -0,0 +1,283 @@
+        )
+
+    @classmethod
+    def process_trades(cls, timestamp: Timestamp, data: List[BinanceCsvRow]) -> List[Trade]:
+        same_assets = True
+        assets: Dict[str, Optional[Asset]] = defaultdict(lambda: None)
+        for row in data:
+            if row['Operation'] == 'Fee':
+                cur_operation = 'Fee'
+            elif row['Change'] < 0:
+                cur_operation = 'Sold'
+            else:
+                cur_operation = 'Bought'
+            assets[cur_operation] = assets[cur_operation] or row['Coin']
+            if assets[cur_operation] != row['Coin']:
+                same_assets = False
+                break
+
+        if not same_assets:
+            for row in data:
+                try:
+                    price = PriceHistorian.query_historical_price(
+                        from_asset=row['Coin'],
+                        to_asset=A_USD,
+                        timestamp=timestamp,
+                    )
+                except NoPriceForGivenTimestamp:
+                    log.warning(f'Couldn\'t find price of {row["Coin"]} on {timestamp}')
+                    return []
+                row['usd_value'] = row['Change'] * price
+
+        def ops_key(x: BinanceCsvRow) -> bool:
+            return x['Operation'] == 'Fee'
+        rows_by_operation = {}
+        for is_fee, rows in groupby(sorted(data, key=ops_key), key=ops_key):
+            rows_by_operation[is_fee] = sorted(
+                rows, key=lambda x: x['Change'] if same_assets else x['usd_value'],
+            )
+
+        trade_rows = []
+        while len(rows_by_operation[False]) > 0:
+            cur_batch = [rows_by_operation[False].pop(0), rows_by_operation[False].pop(-1)]
+            if True in rows_by_operation:
+                cur_batch.append(rows_by_operation[True].pop(0))
+            trade_rows.append(cur_batch)
+
+        prepared_trade_dicts = []
+        for t_rows in trade_rows:
+            cur_data: BinanceCsvRow = defaultdict(lambda: None)
+            for row in t_rows:
+                cur_asset = row['Coin']
+                amount = row['Change']
+                if row['Operation'] == 'Fee':
+                    cur_data['fee_asset'] = cur_asset
+                    cur_data['fee_amount'] = Fee(amount)
+                else:
+                    cur_data['trade_type'] = TradeType.SELL if row['Operation'] == 'Sell' else TradeType.BUY  # noqa:  E501
+                    if amount < 0:
+                        cur_data['from_asset'] = cur_asset
+                        cur_data['from_amount'] = AssetAmount(-amount)
+                    else:
+                        cur_data['to_asset'] = cur_asset
+                        cur_data['to_amount'] = amount
+            prepared_trade_dicts.append(cur_data)
+
+        def trades_key(t_dict: Dict) -> str:
+            return ';'.join(sorted([f'{x[0]}: {x[1]}' for x in t_dict.items()]))
```

Comment:
As before, can this be taken out of here? Also I'm not sure what you have here right now. Could you add comments and maybe with a small example in the docstring?

Reply:
Yep, added!